### DISCUSSION TITLE
Help to implement `wsgidav` with python library

### DISCUSSION THREAD

**Discussion by sventek-s** [mar10/wsgidav#285] [Q&A]
I am trying to implement a webdav server with python library on windows 10. Here is the script that I am using

```py
import os
import logging

from cheroot import wsgi
from wsgidav.wsgidav_app import WsgiDAVApp

# current directory path
cur_dir = os.getcwd()

print(f"[+] Current directory path: {cur_dir}")

config ={
	"host": "0.0.0.0",
	"port": 8080,
	"provider_mapping": {
		"/": cur_dir,
	},
	"verbose": 5,
    # HTTP Authentication Options
    "http_authenticator": {
        "domain_controller": None,
        "accept_basic": False,
        "trusted_auth_header": None,
     },
}

app = WsgiDAVApp(config)

server_args = {
    "bind_addr": (config["host"], config["port"]),
    "wsgi_app": app,
}

# set logging for debugging
#logger = logging.getLogger("wsgidav")
#logger.propagate = True
#logger.setLevel(logging.DEBUG)

server = wsgi.Server(**server_args)
print("[+] Starting server.")
server.start()
print("[+] Started.")
```

Problems I am facing:
- verbose config option is not logging debug info, so I am unable to see anything.
- webhost is accessible from the browser but I am unable to login, its asking for username and password.

How can I fix those two issues, kindly assist. Most of the script I have created from the example/documentation you have provided is there anything I am doing wrong?

Also how does one quit after server.start() is called. When I do a `CTRL + C` I get the below error and I am forced to kill the whole command prompt window.

![image](https://user-images.githubusercontent.com/70489395/233512650-ca0e8ed0-bebc-446b-bd72-0200a6288478.png)

Now is there a way to kill the server like `server.stop()`. I noticed this is a `wsgi` library issue and not `wsgidav` library. But I have been looking for the option to stop it but still unable. Anyways one step at a time. How to fix the first two issues.

Weird thing the command line option works pretty well, but I want to use the library.

![image](https://user-images.githubusercontent.com/70489395/233512826-4449520a-20a6-4246-abda-e25e2c28ed6f.png)

**Comment by mar10** [comment]
Hi,

you could try this pattern to enable logging and <kbd>Ctrl</kbd>-<kbd>C</kbd> handling:

```py
import os

from cheroot import wsgi
from wsgidav.wsgidav_app import WsgiDAVApp
from wsgidav.util import init_logging

# current directory path
cur_dir = os.getcwd()

print(f"[+] Current directory path: {cur_dir}")

config = {
    "host": "0.0.0.0",
    "port": 8080,
    "provider_mapping": {
        "/": cur_dir,
    },
    "verbose": 5,
    "logging": {
        "enable": True,  # @since 4.3.0
    },
    # HTTP Authentication Options
    "http_authenticator": {
        "domain_controller": None,
        "accept_basic": False,
        "trusted_auth_header": None,
    },
}

# Before v4.3.0:
# init_logging(config)

app = WsgiDAVApp(config)

server_args = {
    "bind_addr": (config["host"], config["port"]),
    "wsgi_app": app,
}

server = wsgi.Server(**server_args)

try:
    print("Starting server...")
    server.start()
except KeyboardInterrupt:
    print("Caught Ctrl-C: stopping...")
finally:
    server.stop()
```
